commerce-account-service/src/main/java/com/liferay/commerce/account/service/impl/CommerceAccountLocalServiceImpl.java

```diff
@@ -796,28 +796,28 @@ protected void validate(
 			commerceAccountPersistence.fetchByC_ERC(
 				companyId, externalReferenceCode);
 
-		if ((commerceAccount != null) &&
-			(commerceAccount.getCommerceAccountId() != commerceAccountId)) {
-
-			throw new DuplicateCommerceAccountException(
-				"There is another commerce account with external reference " +
-					"code " + externalReferenceCode);
-		}
+		if (commerceAccount != null) {
+			if (commerceAccount.getCommerceAccountId() != commerceAccountId) {
+				throw new DuplicateCommerceAccountException(
+					"There is another commerce account with external " +
+						"reference code " + externalReferenceCode);
+			}
 
-		if ((commerceAccount.getDefaultBillingAddressId() != 0) &&
```

Comment:
@alecsloan I still not understand the sense of this check. Can you explain to me why we do need it?

Reply:
the check is because of the toggle that we have in CP > Accounts > Address section. It validates that the Account doesn't already have a default x address set.

Reply:
better create a different validator for addresseds, and don't share with uptadate and create method

Reply:
Different validator or update method?